src/python-kolibri-flatpak-utils/run_kolibri_desktop.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/python3
+
+import subprocess
+
+from .kolibri_redirect import KolibriRedirectThread
+from .utils import is_kolibri_socket_open
+
+
+def kolibri_desktop_main(path='/'):
+    if is_kolibri_socket_open():
+        return _run_desktop(path)
+    else:
+        return _run_desktop_and_service(path)
+
+
+def _run_desktop(path):
+    kolibri_redirect = KolibriRedirectThread()
+
+    kolibri_redirect.start()
+
+    kolibri_redirect.await_started()
+
+    kolibri_redirect_url = "http://127.0.0.1:{port}/?next={path}".format(
+        port=kolibri_redirect.redirect_server_port,
+        path=path
+    )
+    result = subprocess.run(['xdg-open', kolibri_redirect_url])
+
+    if result.returncode == 0:
+        kolibri_redirect.join()
+    else:
+        kolibri_redirect.stop()
```

Comment:
It does have documented non-zero exit codes, but the only time I can think of is if the selected browser is unavailable, or some permissions issue is preventing xdg-open from starting it. I kind of like having this check just because it stops us from running for nothing (and the launcher will exit immediately with whatever error xdg-open raised), but on the other hand it's code that will almost never get run ¯\_(ツ)_/¯ I ended up removing the extra code and added some console output that will help to figure things out in the event it doesn't work correctly.